modules/test/playwright/pages/dynamic-data-mapping-form-web/FormBuilderSidePanelPage.ts

```diff
@@ -76,10 +76,25 @@ export class FormBuilderSidePanelPage {
 		);
 	}
 
-	async addFieldByDoubleClick(formFieldTypeTitle: FormFieldTypeTitle) {
+	async addFieldByDoubleClick(
+		formFieldTypeTitle: FormFieldTypeTitle,
+		returnReference?: boolean
+	) {
 		await this.page
 			.getByTitle(formFieldTypeTitle, {exact: true})
 			.dblclick();
+
+		if (returnReference) {
+			await this.page.waitForLoadState('networkidle');
+
+			await this.clickAdvancedTab();
+
+			const fieldReferenceValue = this.page
+				.getByLabel('Field Reference')
+				.inputValue();
+
+			return fieldReferenceValue;
+		}
```

Comment:
I believe it would be better to take this logic into a separate method, like this:
```
async getFieldReference(){
	await this.clickAdvancedTab();

	const fieldReferenceValue = this.page
		.getByLabel('Field Reference')
		.inputValue();

	return fieldReferenceValue;
}
```